spec/service_models/miq_ae_service_orchestration_stack_spec.rb

```diff
@@ -45,4 +46,127 @@
       service_stack.class.refresh(stack.ext_management_system.id, stack.ems_ref)
     end
   end
+
+  it "#start_retirement" do
+    expect(service_stack.retirement_state).to be_nil
+    service_stack.start_retirement
+
+    expect(service_stack.retirement_state).to eq("retiring")
+  end
+
+  it "#retire_now" do
+    expect(stack.retirement_state).to be_nil
+    expect(MiqEvent).to receive(:raise_evm_event).once
+
+    service_stack.retire_now
+  end
+
+  it "#extend_retires_on - invalid date" do
+    error_msg = "Invalid Date specified: #{Time.zone.today}"
+    expect { service_stack.extend_retires_on(7, Time.zone.today) }.to raise_error(RuntimeError, error_msg)
+  end
+
+  it "#finish_retirement" do
+    expect(service_stack).not_to be_retired
+    expect(service_stack.retirement_state).to be_nil
+    expect(service_stack.retires_on).to be_nil
+
+    service_stack.finish_retirement
+
+    expect(service_stack.retired).to be_truthy
+    expect(service_stack.retires_on).to be_between(Time.zone.now - 1.hour, Time.zone.now + 1.second)
+    expect(service_stack.retirement_state).to eq("retired")
+  end
+
+  it "#retiring - false" do
+    expect(service_stack.retiring?).to be_falsey
+  end
+
+  it "#retiring? - true" do
+    service_stack.retirement_state = 'retiring'
+
+    expect(service_stack.retiring?).to be_truthy
+  end
+
+  it "#error_retiring? - false" do
+    expect(service_stack.error_retiring?).to be_falsey
+  end
+
+  it "#error_retiring? - true" do
+    service_stack.retirement_state = 'error'
+
+    expect(service_stack.error_retiring?).to be_truthy
+  end
+
+  it "#retires_on - today" do
+    stack.update_attributes(:retirement_last_warn => Time.zone.now)
+    service_stack.retires_on = Time.zone.now
+    stack.reload
+    expect(stack.retirement_last_warn).to be_nil
+    expect(stack.retirement_due?).to be_truthy
+  end
+
+  it "#retires_on - tomorrow" do
+    stack.update_attributes(
+      :retired              => true,
```

Comment:
false

Reply:
@jrafanie The description is incorrect for this test. I should have described it better when I initially added the test.  I'll fix that in the other tests. :-)  
The test is intended to make sure that those retirement fields are cleared when the retires_on date is changed.  @d-m-u Can you change the description here?

Reply:
ah, thanks... can the retired field be nil and reset during the test?